### ISSUE TITLE
upload successfully built staged-recipes to a conda-forge-alpha channel

### ISSUE THREAD

**Issue by notestaff** [conda-forge/conda-forge.github.io#464]
@conda-forge/core Could successfully built staged-recipes be uploaded to a separate anaconda.org channel, say conda-forge-alpha? Then they can be used sooner by people who are ok with the risk. One can add conda-forge-alpha at a lower channel priority than conda-forge, so it would not mask any official packages.

I know I can build packages locally and upload them to my own channel, but the build process won't exactly replicate conda-forge's, so the resulting package won't be the same. If something works with that package, it may not work with the eventual "official" one. And sometimes my local builds fail for unclear reasons, while the conda-forge build succeeds. Also, I don't have a Mac, so can't make a local package to share with colleagues who have Macs.

**Comment by notestaff** [comment]
(Issue moved here from conda-forge/staged-recipes#4155 per @jakirkham 's suggestion)

**Comment by notestaff** [comment]
Also, Linux builds often finish long before others even start, so it'd help to have successful Linux builds be available quickly (for when I'm adding a package that I want to use for my own work).  Also, it can take longer to get Mac or Windows builds working at all; it'd be good if that did not hold up availability (in alpha) of successful Linux builds.

**Comment by isuruf** [comment]
I think it's pretty easy to create non-conda-forge feedstocks that use the infrastructure of conda-forge. See https://github.com/conda-forge/conda-smithy#making-a-new-feedstock

**Comment by notestaff** [comment]
"it's pretty easy to create non-conda-forge feedstocks that use the infrastructure of conda-forge" -- thanks, good to know.  This helps for recipes I submit.  There's still the case of using others' successfully built staged-recipes before they get into conda-forge (which sometimes takes a couple of weeks).  Also, staged-recipes could reference other staged-recipes as dependencies, if conda-forge-alpha was added as a low-priority channel when staged-recipes were tested.   Right now the only way is to submit several recipes in one PR.  But such PRs can take longer to review.  E.g. right now I have a staged-recipes recipe for git-annex, which required also adding a recipe for rsync, which required also adding a recipe for popt.  Ideally I could submit three smaller PRs which could then be taken up in smaller chunks by reviewers.

**Comment by notestaff** [comment]
One more issue with requiring multi-recipe PRs is that the CI test time adds up, and also you end up re-running the tests for the dependencies of your main package, even if their recipes are already debugged.

**Comment by ChrisBarker-NOAA** [comment]
I like the idea of an Alpha, or temp channel.

It would be a way t o access and test the actual package as built by conda-forge before making into the proper channel.

In theory, the in-line tests should cover that, but we all know about "complete" test coverage.

And there is no way the inline tests could be testing how it interacts with any number of other packages.

**Comment by mariusvniekerk** [comment]
Technically this could also work as a place to put packages built during feedstock pull requests?

**Comment by jakirkham** [comment]
We had been discussing something like that before.

ref: https://paper.dropbox.com/doc/Staged-Releases-r9My2gvS5vb2VMIlf3xue

**Comment by notestaff** [comment]
"this could also work as a place to put packages built during feedstock pull requests" -- that would be useful, because then one can submit new recipes dependent on newer version of a feedstock, and have them build properly before the feedstock PR is merged (which can take a while).

**Comment by notestaff** [comment]
Some PRs now take over a month to get merged, e.g. https://github.com/conda-forge/staged-recipes/pull/4150
An alpha channel would  mitigate this problem.

**Comment by scopatz** [comment]
Hi @notestaff - I sympathize with the problem you have identified. However, I think the solution is to have more people who are willing and able to review staged-recipes, and do so on a semi-regular basis. I think we'd welcome anyone who wanted to help out in this way.  I am not sure that an alpha channel would actually mitigate the underlying problem.

**Comment by notestaff** [comment]
@scopatz What an alpha channel would do is let me quickly build and release software that assumes the existence of a package in conda-forge, and not have to change the software when the package is eventually added.  I can tell my users that, to be able to install and use my software, they must add the conda-forge-alpha channel (at a lower priority than conda-forge).  Then my releases are not held up by the conda-forge review cycle.  Obviously, my users are then accepting some risk from relying on non-reviewed packages, but they do it knowingly.

What would be the downsides of having an alpha channel?

**Comment by ChrisBarker-NOAA** [comment]
I handle that by having my own channel, and I have my users put it on the list after conda-forge.

Then when they install, it looks frist n conda forge, and if it's there, it gets used. if not, it'll find it on my provate channel.

So I push things first to the private channel when I'm ready to release, by users get to use it, and then, when it finally makes it into conda-forge, they'll just start getting it there.

This has the advantage that my users (and me!) are only getting "alpha" packages from our group, and not everyone else's unvetted stuff.

It means you need to maintain yur own channel, but that's actually pretty lightweight -- and you have to test the whole build process somewhere anyway.

-CHB
